tests/agents/test_dynamic_bounds_and_handoff.py

```
@@ -0,0 +1,91 @@
+        ],
+        [AIMessage(content="m4")],
+    )
+
+    assert isinstance(result[0], SystemMessage)
+    assert result[0].name == "server_memory_context"
+    visible = [msg.content for msg in result[1:]]
+    assert visible == ["m3", "m4"]
+
+
+@pytest.mark.asyncio
+async def test_handoff_enriched_payload_adds_expected_fields(monkeypatch) -> None:
```

Ruff (0.14.14)

[warning] Unused function argument: `monkeypatch`

(ARG001)